uint8 mode
int16 x
int16 y
uint8 stop
